index.js

```diff
@@ -92,18 +92,41 @@ function realClickInternal(name, options, parent, window, renderer, done) {
 
 // Utilities
 
-function actionOnElementCenter(actionName) {
-  return function(selector, done) {
+function actionOnElement(actionName) {
+  return function(selector /*[, position], done */) {
+    const done = arguments[arguments.length - 1]
+    const position = arguments.length > 2 ? arguments[1] : undefined;
     if (typeof selector !== 'string') {
       return done(new TypeError(`${actionName}: "selector" must be a string`));
     }
-    debug(`Finding "${selector}"`);
     var child = this.child;
-    getBounds(this, selector).then(center).then(function(point) {
+
+    debug(`Finding "${selector}"`);
+    getBounds(this, selector).then(function calcPosition(bounds) {
+      if (position) {
+        if (typeof position.x !== 'number' || typeof position.y !== 'number') {
+          throw new TypeError('x and y must be numbers')
+        } else if (
+            position.x < bounds.left  ||
+            bounds.right < position.x ||
+            position.y < bounds.top   ||
+            bounds.bottom < position.y
```

Comment:
Hmmm, this check doesn’t seem right. If `position` is *relative* to the element’s bounds, don’t you need to check it *after* adding it to the bounds?

e.g. If I have a button with bounds `{left: 10, top: 10, width: 100, height: 10}` and call:

```js
nightmareInstance.realClick('button', {x: 1, y: 1})
```

That will throw an error because of this test.

Maybe a good idea to ensure that the `should accept specified position` covers this case?

Reply:
You're REALLY careful!